test/src/system/block_formatting_test.coffee

```diff
@@ -279,3 +279,76 @@ testGroup "Block formatting", template: "editor_empty", ->
       clickToolbarButton attribute: "number", ->
         assert.blockAttributes([0, 1], ["numberList", "number"])
         done()
+
+  test "adding bullet to heading block", (done) ->
+    clickToolbarButton attribute: "heading1", ->
+      clickToolbarButton attribute: "bullet", ->
+        assert.ok isToolbarButtonActive(attribute: "heading1")
+        assert.blockAttributes([1, 2], [])
+        done()
+
+  test "removing bullet from heading block", (done) ->
+    clickToolbarButton attribute: "bullet", ->
+      clickToolbarButton attribute: "heading1", ->
+        assert.ok isToolbarButtonDisabled(attribute: "bullet")
+        done()
+
+  test "breaking out of heading in list", (done) ->
+    clickToolbarButton attribute: "bullet", ->
+      clickToolbarButton attribute: "heading1", ->
+        assert.ok isToolbarButtonActive(attribute: "heading1")
+        typeCharacters "abc", ->
+          typeCharacters "\n", ->
+            assert.ok isToolbarButtonActive(attribute: "bullet")
+            document = getDocument()
+            assert.equal document.getBlockCount(), 2
+
+            block = document.getBlockAtIndex(0)
+            assert.deepEqual block.getAttributes(), ["bulletList", "bullet", "heading1"]
+            assert.equal block.toString(), "abc\n"
```

Comment:
Instead of testing each block's string, you can test the whole document. The test callback accepts an optional string and performs an assertion when present:
```coffee
test "breaking out of heading in list", (expectDocument) ->
  ...
  expectDocument("abc\n")
```